python/cugraph/cugraph/gnn/graph_store.py

```diff
@@ -210,67 +211,95 @@ def sample_neighbors(
-        # We filter by creating a series of destination nodes
-        # corresponding to the offsets and filtering non matching vallues
+        if edge_dir not in ["in", "out"]:
+            raise ValueError(
+                f"edge_dir must be either 'in' or 'out' got {edge_dir} instead"
+            )
 
-        seeds_offsets_s = cudf.Series(seeds_offsets).values
-        offset_lens = seeds_offsets_s[1:] - seeds_offsets_s[0:-1]
-        dst_seeds = current_seeds.repeat(offset_lens)
-        dst_seeds.index = ego_edge_list.index
-        filtered_list = ego_edge_list[ego_edge_list["dst"] == dst_seeds]
+        nodes = cudf.from_dlpack(nodes).astype(cp.int32)
 
-        del dst_seeds, offset_lens, seeds_offsets_s
-        del ego_edge_list, seeds_offsets
+        if edge_dir == "in":
+            sg = self.extracted_reverse_subgraph_without_renumbering
+        else:
+            sg = self.extracted_subgraph_without_renumbering
 
-        # Step 2
-        # Sample Fan Out
-        # for each dst take maximum of fanout samples
-        filtered_list = sample_groups(
-            filtered_list, by="dst", n_samples=fanout
+        sampled_df = uniform_neighbor_sample(
+            sg, start_list=nodes, fanout_vals=[fanout],
+            with_replacement=replace
         )
 
-        # TODO: Verify order of execution
-        sample_df = cudf.DataFrame(
-            {src_n: filtered_list["src"], dst_n: filtered_list["dst"]}
-        )
-        del filtered_list
+        sampled_df.drop(columns=["indices"], inplace=True)
 
-        # del parents_nodes, children_nodes
-        edge_df = sample_df.merge(
-            self.gdata._edge_prop_dataframe[[src_n, dst_n, eid_n]],
-            on=[src_n, dst_n],
-        )
+        # handle empty graph case
+        if len(sampled_df) == 0:
+            return None, None, None
+
+        # we reverse directions when directions=='in'
+        if edge_dir == "in":
+            sampled_df.rename(
+                columns={"destinations": src_n, "sources": dst_n}, inplace=True
+            )
+        else:
+            sampled_df.rename(
+                columns={"sources": src_n, "destinations": dst_n}, inplace=True
+            )
+
+        edge_df = self.gdata._edge_prop_dataframe[[src_n, dst_n, eid_n]]
+        sampled_df = edge_df.merge(sampled_df)
 
         return (
-            edge_df[src_n].to_dlpack(),
-            edge_df[dst_n].to_dlpack(),
-            edge_df[eid_n].to_dlpack(),
+            sampled_df[src_n].to_dlpack(),
+            sampled_df[dst_n].to_dlpack(),
+            sampled_df[eid_n].to_dlpack(),
         )
 
-    def find_edges(self, edge_ids, etype):
+    @cached_property
+    def extracted_reverse_subgraph_without_renumbering(self):
+        # TODO: Switch to extract_subgraph based on response on
+        # https://github.com/rapidsai/cugraph/issues/2458
+        subset_df = self.gdata._edge_prop_dataframe[[src_n, dst_n]]
```

Comment:
Can you add a FIXME to use `PropertyGraph.get_edge_data()` when #2444 is merged since we're trying to remove exposing the internal PG dataframes?

Reply:
I have added all the relevant links to issues as needed.  Let me know if we want to link anything else.